api_types.ts

```diff
@@ -893,6 +954,15 @@ export interface ExecutionContext {
    * TODO(patrick): Unhide this
    */
   readonly executionId?: string;
+
+  /**
+   * If this invocation is a retry, this will be populated with information about what went wrong
+   * during the previous attempt.
+   *
+   * TODO(patrick): Unhide this
+   * @hidden
+   */
+  readonly previousAttemptError?: InvocationError;
```

Comment:
I guess ideally we'd put this in `Sync` since that's the only place where "previous" makes any sense, but maybe that's not possible due to desired usage for executeGetPermissions or something?

Reply:
We actually put our auto-retry logic into other places too, like getSchema

Reply:
so this is a different "previous" than a sync's "previous"